homeassistant/components/smarttub/climate.py

```diff
@@ -0,0 +1,119 @@
+"""Platform for climate integration."""
+import logging
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_HEAT,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.util import slugify
+from homeassistant.util.temperature import convert as convert_temperature
+
+from .const import DEFAULT_MAX_TEMP, DEFAULT_MIN_TEMP, DOMAIN, SMARTTUB_CONTROLLER
+from .entity import SmartTubEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up climate entity for the thermostat in the tub."""
+
+    controller = hass.data[DOMAIN][entry.entry_id][SMARTTUB_CONTROLLER]
+
+    entities = [
+        SmartTubThermostat(controller.coordinator, spa) for spa in controller.spas
+    ]
+
+    async_add_entities(entities)
+
+    return True
+
+
+class SmartTubThermostat(SmartTubEntity, ClimateEntity):
+    """The target water temperature for the spa."""
+
+    def __init__(self, coordinator, spa):
+        """Initialize the entity."""
+        super().__init__(coordinator, spa, "thermostat")
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique id for the entity."""
+        return f"{self.spa.id}-{slugify(self._entity_name)}"
```

Comment:
Can the `entity_name` be changed?  We are setting this value so we shouldn't have to slugify it.

Ideally we don't want to have `name` in something with a unique id as it implies it could change.  `self._entity_type` would be cleaner here